### PR TITLE
fix: first-run correctness & documentation-drift pass (11 issues)

### LINKED ISSUE #107: `pip install skdr-eval[choice]` is documented but the `[choice]` extra is not defined
## Newcomer impact

A newcomer following the README's "Installation → Optional Dependencies" runs the command and pip silently warns `does not provide the extra 'choice'`. They believe they enabled conditional-logit propensity support; they didn't, because no such extra exists.

## Evidence

- README: `README.md:89-91` documents `pip install skdr-eval[choice]`.
- Preflight docstring: `examples/preflight.py:6` references `[choice]` too.
- `pyproject.toml:59-91` defines `speed`, `viz`, `dev`, `examples`, `notebooks` — but **not** `choice`.

Reproduction:

```
$ pip install 'skdr-eval[choice]'
WARNING: skdr-eval 0.8.0 does not provide the extra 'choice'
```

Expected: either the `[choice]` extra is defined (pulling in any dependency required by `src/skdr_eval/choice.py`), or the docs are corrected.

## Steps to reproduce

1. `pip install 'skdr-eval[choice]'`
2. See the pip warning in stderr.

## Proposed fix

- Inspect `src/skdr_eval/choice.py` to determine which dependencies (if any) are needed beyond core.
- If none are needed (conditional-logit uses only existing deps): remove the `[choice]` line from `README.md` and `examples/preflight.py` docstring; audit `docs/agent-context/` for other references.
- If a dependency is needed: add `[project.optional-dependencies] choice = [...]` and document what it unlocks.

## Implementation notes

- Likely files: `pyproject.toml`, `README.md`, `examples/preflight.py`.
- Tests to add: `tests/test_capabilities.py` — assert that every documented extra in README corresponds to a declared extra in `pyproject.toml` (guards future drift).
- Verify `get_capabilities()` does not silently report `'choice'` as missing if the extra is removed.

## Acceptance criteria

- [ ] `pip install 'skdr-eval[choice]'` does not emit "does not provide the extra", OR README/preflight no longer reference it.
- [ ] `examples/preflight.py` docstring is consistent with `pyproject.toml`.
- [ ] A test guards against future drift between declared and documented extras.

## Duplicate check

Searched open issues for: "choice extra", "optional-dependencies", "pip install extra". Result: no open duplicate found.

### LINKED ISSUE #109: `evaluate_sklearn_models(models=<single_estimator>)` raises a cryptic `AttributeError` instead of validating the dict contract
## Newcomer impact

Single-model evaluation is the natural starting point for a new user. Passing the estimator directly (instead of a `{name: estimator}` dict) currently tracebacks deep inside the internal pipeline at `estimators_` access. The error message gives no hint that the user simply needs to wrap their model in a dict.

## Evidence

- Entry point: `src/skdr_eval/core.py` (`evaluate_sklearn_models` and `evaluate_pairwise_models`).

Reproduction:

```python
from sklearn.ensemble import RandomForestRegressor
import skdr_eval

logs, _, _ = skdr_eval.make_synth_logs(n=200, n_ops=3, seed=42)
skdr_eval.evaluate_sklearn_models(
    logs=logs,
    models=RandomForestRegressor(random_state=42),  # bug: should be a dict
    fit_models=True,
    n_splits=3,
    policy_train="pre_split",
)
```

- Actual: `AttributeError: 'RandomForestRegressor' object has no attribute 'estimators_'`
- Expected: `DataValidationError: 'models' must be a dict of {name: estimator}. Got <RandomForestRegressor>; did you mean models={"rf": <RandomForestRegressor>}?`

## Steps to reproduce

1. Call `evaluate_sklearn_models(logs=..., models=RandomForestRegressor(), fit_models=True, n_splits=3)`.
2. Observe the internal `AttributeError`.

## Proposed fix

- At the top of `evaluate_sklearn_models` (and `evaluate_pairwise_models`), check `isinstance(models, dict)`. If not, raise `DataValidationError` with the actionable message above. Same for empty dict, non-string keys, non-estimator values.
- Add a small `validate_models_dict` helper in `validation.py` so both entry points share the check.

## Implementation notes

- Likely files: `src/skdr_eval/core.py` (both entry points), `src/skdr_eval/validation.py` (new helper).
- Tests to add: `tests/test_api.py` — assert raises on `models=<estimator>`, `models={}`, `models={"x": 42}`.
- Backward compatibility: only error path changes; valid calls unaffected.

## Acceptance criteria

- [ ] Passing a single estimator raises `DataValidationError` with a "did you mean…?" hint.
- [ ] Passing `{}` raises with "models dict is empty".
- [ ] Passing `{"name": <non-estimator>}` raises with a clear type error.
- [ ] New tests cover all three regressions.

## Duplicate check

Searched open issues for: "single model", "models dict", "AttributeError estimators_". Result: no open duplicate found.

### LINKED ISSUE #110: README "Trusted Publishing Setup" lists the wrong repository owner (`dandrsantos/skdr-eval`)
## Newcomer impact

A release engineer following the README's Trusted Publishing setup will configure the wrong GitHub repo on PyPI, breaking auto-release.

## Evidence

- File/line: `README.md:471`
- Actual content: `**Repository**: dandrsantos/skdr-eval`
- Expected: `dgenio/skdr-eval` (matches every other repo reference in the README and `pyproject.toml`).

```
$ grep -n dandrsantos README.md
471:   - **Repository**: `dandrsantos/skdr-eval`
```

## Steps to reproduce

1. `grep -n dandrsantos README.md`

## Proposed fix

- Replace `dandrsantos/skdr-eval` with `dgenio/skdr-eval` in `README.md:471`.

## Implementation notes

- Files to change: `README.md` only.

## Acceptance criteria

- [ ] `grep -n dandrsantos README.md` returns no results.

## Duplicate check

Searched open issues for: "dandrsantos", "Trusted Publishing". Result: open #98 mentions an (older) claim that `pyproject.toml` references `dandrsantos`. In fact `pyproject.toml` is already correct; the remaining occurrence is *only* this README line. This issue is narrower and orthogonal.

## Existing related issues

- #98 *Fix repository metadata URLs and clarify the practical use case* — could be cross-linked. Its `pyproject.toml` premise is no longer accurate; only the README line covered here remains.

### LINKED ISSUE #111: README citation block shows stale `version = {0.7.0}` while CITATION.cff/CHANGELOG list 0.8.0
## Newcomer impact

Academic users copy-paste the BibTeX block out of the README; they end up citing an out-of-date version.

## Evidence

- `README.md:485`: `version = {0.7.0}`
- `CITATION.cff:5`: `version: "0.8.0"`
- `CHANGELOG.md:65`: `## [0.8.0] - 2026-05-20`
- Live PyPI latest: `0.8.0`

## Steps to reproduce

1. `grep -n "version = " README.md`
2. Compare with `head -10 CITATION.cff`.

## Proposed fix

- Bump README's BibTeX `version` field to `0.8.0`.
- Optional: drive the version dynamically in `scripts/create_release.py` so the README stays in sync on every release (`sed`-bump or a Jinja template).

## Implementation notes

- Likely files: `README.md`; optionally `scripts/create_release.py`.
- Tests: optional doc-consistency unit test asserting `README.md` BibTeX version matches `CITATION.cff` version.

## Acceptance criteria

- [ ] README BibTeX `version = {0.8.0}`.
- [ ] (Optional) test guards against future drift.

## Duplicate check

Searched open issues for: "version 0.7.0 stale", "CITATION.cff README". Result: no direct duplicate. Related: #77 covers Zenodo DOI / CITATION.cff polish more broadly but does not call out the README BibTeX block.

### LINKED ISSUE #112: CONTRIBUTING.md says "Python 3.9+" but `pyproject.toml` requires `>=3.11`
## Newcomer impact

A would-be contributor with Python 3.9 follows `CONTRIBUTING.md`, fails install, debugs for ten minutes before realizing the real minimum is 3.11.

## Evidence

- `CONTRIBUTING.md:149`: `- Python 3.9+`
- Source of truth: `pyproject.toml:48` → `requires-python = ">=3.11"`

## Steps to reproduce

1. `grep -n "Python 3" CONTRIBUTING.md`
2. Compare with `requires-python` in `pyproject.toml`.

## Proposed fix

- Change `Python 3.9+` to `Python 3.11+` in `CONTRIBUTING.md:149`.
- Audit `DEVELOPMENT.md` and `docs/DEVELOPMENT.md` for the same drift.

## Implementation notes

- Likely files: `CONTRIBUTING.md`, `DEVELOPMENT.md`, `docs/DEVELOPMENT.md`.
- Tests: none required.

## Acceptance criteria

- [ ] `grep -nE "Python 3\.9|Python 3\.10" CONTRIBUTING.md DEVELOPMENT.md docs/DEVELOPMENT.md` returns no matches that contradict `pyproject.toml`.

## Duplicate check

Searched open issues for: "Python 3.9", "Python version", "CONTRIBUTING.md". Result: no open duplicate found.

### LINKED ISSUE #113: `make_pairwise_synth` docstring example shows wrong DataFrame shapes
## Newcomer impact

A user reading the docstring example sees `(300, 12)` / `(30, 5)` but actual output is `(300, 14)` / `(30, 6)`. Small drifts like this erode trust in the docs.

## Evidence

- File: `src/skdr_eval/synth.py:202-206`
- Docstring claim:

```
>>> print(logs_df.shape)
(300, 12)  # 3 days * 100 clients, ~12 columns
>>> print(op_daily_df.shape)
(30, 5)    # 3 days * 10 ops, ~5 columns
```

- Actual:

```
$ python -c "import skdr_eval; a,b = skdr_eval.make_pairwise_synth(n_days=3, n_clients_day=100, n_ops=10); print(a.shape, b.shape)"
(300, 14) (30, 6)
```

## Steps to reproduce

1. Run the one-liner above.
2. Compare with the docstring.

## Proposed fix

- Update the docstring to match the current output (or omit hard-coded shape numbers).
- Better: convert the example to a real doctest that CI runs, so this can never drift again.

## Implementation notes

- Likely files: `src/skdr_eval/synth.py`.
- Tests: enable `--doctest-modules` for `synth.py`, or add a focused test asserting shape stability.

## Acceptance criteria

- [ ] Docstring shapes match reality.
- [ ] (Optional) doctest enforces it.

## Duplicate check

Searched open issues for: "make_pairwise_synth", "synth shape", "doctest". Result: no open duplicate found.

### LINKED ISSUE #115: `validate_logs` action-vs-eligibility error is unreadable when `action` is int but `*_elig` column names parse as str
## Newcomer impact

A common newcomer mistake is encoding actions as integer indices (`0, 1, 2`) while the eligibility columns are named `0_elig`, `1_elig`, `2_elig`. The validator's error message prints two identical-looking lists, with nothing to tell the user the difference is `int` vs `str`.

## Evidence

- File: `src/skdr_eval/validation.py:465-470`

Reproduction:

```python
import skdr_eval

logs = skdr_eval.make_synth_logs(n=100, n_ops=3, seed=42)[0]
logs["action"] = logs["action"].map({"op_A": 0, "op_B": 1, "op_C": 2})
logs = logs.rename(columns={"op_A_elig": "0_elig", "op_B_elig": "1_elig", "op_C_elig": "2_elig"})
skdr_eval.validate_logs(logs)
```

- Actual: `DataValidationError: logs.action contains values not present as '*_elig' columns: [0, 1, 2] (eligible operators: ['0', '1', '2'])`
- Expected: include dtypes in the message, e.g., `... action dtype is int64 but '*_elig' columns parse as str. Cast action to str (logs["action"] = logs["action"].astype(str)) or rename the elig columns.`

## Steps to reproduce

See snippet above.

## Proposed fix

- In `validate_logs`, when `invalid_actions` is non-empty, also include `logs["action"].dtype` and the inferred type of `ops_all` (parsed from `*_elig` column names). Append both to the error message and suggest a cast.

## Implementation notes

- Likely files: `src/skdr_eval/validation.py:465-470`.
- Tests to add: a regression test asserting the dtype hint appears when the int/str mismatch is the cause.
- Edge cases: mixed action dtypes (object column with both ints and strings); already-string actions whose values simply don't match (existing message stays fine for that case).

## Acceptance criteria

- [ ] When action dtype differs from the implicit elig-name dtype, the message says so explicitly.
- [ ] When values are simply unrelated (e.g., action `"op_X"`, elig has `op_A`/`op_B`), message is unchanged.
- [ ] New test covers the int/str mismatch path.

## Duplicate check

Searched open issues for: "validate_logs error", "action dtype", "elig mismatch". Result: no open duplicate found.

### PR DESCRIPTION
**Pull request by dgenio**
# Pull Request

## 📋 Description

A single coherent pass over the **first-run correctness & documentation-drift** issues opened on 2026-05-21. Each fix addresses a place where *documented behavior did not match the code*, plus the small public-API contract gaps those mismatches exposed. The changes are individually tiny and share one implementation path (README/CONTRIBUTING/pyproject + a handful of validation/serialization tweaks), so they are cleaner to review and ship together than as 11 separate PRs.

### Type of Change
- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (see **API Changes** — return-type and exception-type adjustments; intentional, no back-compat shims per maintainer direction)
- [x] 📚 Documentation update
- [ ] 🧹 Code refactoring (no functional changes)
- [ ] ⚡ Performance improvement
- [x] 🧪 Test improvements
- [ ] 🔧 Build/CI improvements

## 🔗 Related Issues

- Fixes #104
- Fixes #105
- Fixes #107
- Fixes #108
- Fixes #109
- Fixes #110
- Fixes #111
- Fixes #112
- Fixes #113
- Fixes #114
- Fixes #115

### Test Coverage
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added integration tests if applicable — N/A (unit-level fixes)

### Manual Testing
- [x] Tested locally (`ruff`, `mypy`, full `pytest` with coverage)
- [x] Reproduced every bug before the fix and confirmed the fix resolves it
- [x] Tested edge cases (custom `y_col`, empty/non-dict `models`, `policy_train="all"` path left unchanged, unrelated-string actions keep original message)

**Test commands run:**
```bash
ruff check src/ tests/ examples/      # All checks passed!
ruff format --check src/ tests/ examples/   # 80 files already formatted
mypy src/skdr_eval/                   # Success: no issues found in 32 source files
pytest -q                             # 700 passed, 3 skipped — TOTAL coverage 93%
```

### Code Changes
- **#105** `evaluate_sklearn_models` / `build_design` / `validate_logs` accept `y_col` (default `"service_time"`) so general-purpose OPE logs can name the reward column anything (`reward`, `click`, `revenue`).
- **#109** New `validate_models_dict()` in `validation.py`, called by both entry points: raises an actionable `DataValidationError` (with a *"did you mean `models={...}`?"* hint) for a bare estimator, empty dict, non-string keys, or non-estimator values — instead of a deep `AttributeError`.
- **#108** `EvaluationArtifact.to_json()` / `.to_html()` now return the serialized **string** when called with no argument and write the file (returning its `Path`) when given a `path` (matching the existing `EvaluationCard.to_json` idiom).
- **#114** With `policy_train="pre_split"`, the `InsufficientDataError` now reports the original input row count and `policy_train_frac` (was the opaque post-split count). The `policy_train="all"` message is left byte-for-byte unchanged.
- **#115** `validate_logs` adds a dtype-cast hint when `action` is int but `*_elig` column names parse as str; genuinely-unrelated string actions keep the original message.
- **#113** `make_pairwise_synth` docstring shapes corrected to `(300, 14)` / `(30, 6)`.

### Documentation
- **#104** README quickstarts (and the other runnable snippets) pass `policy_train="pre_split"` + a one-line explanation, so a copy-pasted example runs without a `DeprecationWarning`.
- **#107** Removed the undefined `[choice]` extra from README + `examples/preflight.py`; conditional logit is built-in (SciPy is a core dependency).
- **#110** README Trusted-Publishing owner → `dgenio/skdr-eval`.
- **#111** README BibTeX `version` → `0.9.0` (matches `CITATION.cff`).
- **#112** `CONTRIBUTING.md` / `DEVELOPMENT.md` → Python 3.11+ (matches `requires-python`).

### API Changes
- [ ] No API changes
- [x] Backward compatible API additions (`y_col` keyword; both default to prior behavior)
- [x] Breaking API changes (intentional; no compatibility shims per maintainer direction):
  - `to_json()` / `to_html()` with **no argument** now return a `str` (previously raised `TypeError`); calling with a `path` is unchanged (writes + returns `Path`).
  - Empty / `None`-valued `models` now raise `DataValidationError` (was `ValueError`). Existing `test_api.py` assertions were updated accordingly.

## 📚 Documentation
- [x] I have updated the documentation accordingly
- [x] I have updated docstrings for new/modified functions
- [x] I have added examples if applicable (README `y_col` usage, calling conventions)
- [x] I have updated the CHANGELOG.md

### Code Quality
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have run `ruff check` and `ruff format`
- [x] I have run `mypy` type checking

### Testing & CI
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally (700 passed, 3 skipped)
- [ ] All CI checks pass — pending CI on this PR
- [x] Code coverage is maintained or improved (93% total, gate is 80%)

### Documentation & Communication
- [x] I have made corresponding changes to the documentation
- [x] My commit messages follow the conventional commit format
- [x] I have updated the CHANGELOG.md

### Focus Areas
- **#108 return type** is now `str | Path`. `export()` (which always writes a path) casts to `Path`; all internal callers (`cli.py`, tests) pass a path and are unaffected. The issue text suggested a *keyword-only* `path`, but that would break the positional callers — I followed the repo's own `EvaluationCard.to_json(path=None)` idiom instead (documented in the commit).
- **#114** is scoped to `evaluate_sklearn_models` (the path the issue reproduces). The pairwise entry point computes its eval slice differently; enriching it similarly is a sensible follow-up but out of scope here.

### Questions for Reviewers
- **#107**: I removed the `[choice]` docs (SciPy is core, so no extra is needed). If you'd rather *define* a `[choice]` extra instead, say so and I'll switch.

---

**Additional Context:** This group was selected by an issue-triage pass as the most coherent single-PR cluster; six already-implemented issues (#69/#86/#89/#90/#91/#93) were closed separately during that triage.

---
_Generated by [Claude Code](https://claude.ai/code/session_015d9gz1gB62tcVt5TEDoF4Z)_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -628,7 +641,7 @@ If you use this software in your research, please cite:
   author  = {Santos, Diogo},
   year    = {2026},
   url     = {https://github.com/dgenio/skdr-eval},
-  version = {0.7.0},
+  version = {0.9.0},
   license = {MIT}
```
[the remark this answers is not in the document]
Comment (dgenio) [reply README.md:659]: Good catch — fixed in 39eabdd. Bumped `preferred-citation.version` in `CITATION.cff` from `"0.8.0"` to `"0.9.0"` so it matches the top-level `version` (and the README BibTeX block). All three citation references are now aligned on `0.9.0`. Treating top-level `CITATION.cff:version` as the single source of truth for the BibTeX rendering.

---
_Generated by [Claude Code](https://claude.ai/code)_
